### ISSUE TITLE
Report template

### ISSUE THREAD

**Issue by noraj** [reconmap/reconmap#45]
[cbk914](https://github.com/cbk914) (https://github.com/reconmap/reconmap/issues/38#issuecomment-888309091)

> The demo site is not working, there's some way to import the report templates and the vulnerabilities from Github into the Reconmap WUI?

Vulnerability templates should be importable from https://demo.reconmap.org/system/import-data but you need to be connected as admin.

But it's true that report templates are not very configurable.

The documentation https://reconmap.org/user-manual/pentest-report-configuration.html says

> Reconmap simplifies by creation of pentest reports with the use of customisable templates. It comes with a pre-defined pentest report template but you can change styles, cover, header and footer, as well as deciding which options to include or exclude from the report.

It's true that the Configuration tab on Report generation let you choose some options and custom headers for example.

![image](https://user-images.githubusercontent.com/16578570/128221194-07091b81-b074-40f7-951d-b63acc1d4dc1.png)

But the documentation don't say what is the expected format? HTML? a templating language? LaTeX? only text or markdown is supported?

I was expecting something similar to Pwndoc https://pwndoc.github.io/pwndoc/#/docxtemplate or PCF https://gitlab.com/invuls/pentest-projects/pcf/-/wikis/Reports%20moderation

- [x] The ability to import custom report templates
- [x] Template using fiels so the report is 100% customizable
- [ ] The ability to create custom fields in vulnerabilities that can be added in the report

**Comment by noraj** [comment]
unstale

**Comment by santiagolizardo** [comment]
Hi @noraj ; I have implemented a basic report generation system based on Docx templates. You can see it live on the demo instance.
Basically, Reconmap comes with a [very basic Docx template](https://github.com/reconmap/rest-api/blob/master/resources/templates/reports/default.docx), but you can modify it and upload a new one to use in your reports.
![image](https://user-images.githubusercontent.com/220190/134589633-2c8cdabc-6fcb-489d-9c2e-06597afa5a17.png)

![image](https://user-images.githubusercontent.com/220190/134589761-d0f4e486-1259-4b53-98fe-dbf0d79850f0.png)

Hope you find it useful. Feedback as always is very welcome

**Comment by noraj** [comment]
@santiagolizardo Amazing I need to test that soon, but I didn't find the documentation about it? I suppose a pointer to the library used would be enough if the library used already has a documentation explaining how to craft templates.

**Comment by santiagolizardo** [comment]
Good point @noraj . I am using https://phpword.readthedocs.io/en/latest/templates-processing.html to generate the word documents using templates. The syntax is somehow limited (eg no conditionals), but does the trick for now.
Thanks

**Comment by noraj** [comment]
https://github.com/reconmap/website-org/pull/7

**Comment by noraj** [comment]
I checked the doc and the template, so there are no loops but [cloneblocks](https://phpword.readthedocs.io/en/latest/templates-processing.html#cloneblock) to iterate over all vulnerabilities of a project?

**Comment by noraj** [comment]
Seems that vuldash (php) was using PhpWord (php) too https://github.com/talsoft/vuldash/blob/408e44ecfccb96fc6a659f01e4c9e5a05c247d11/application/controllers/Reports.php while Smersh (php) is using Docxtemplater (JS) https://github.com/CMEPW/Smersh/blob/16fe8be4c3f840db41e8650162cfb1e0afd60acf/client/src/app/components/mission-single/mission-single.component.ts and pwndoc (JS) is using Docxtemplater (JS) too https://github.com/pwndoc/pwndoc/blob/6c3ac8d9cd57dbbc17ae2311be955ad925223310/backend/src/routes/template.js and https://github.com/pwndoc/pwndoc/blob/6c3ac8d9cd57dbbc17ae2311be955ad925223310/docs/docxtemplate.md.
While Docxtemplater seems more powerful, it has ended in a freemium open-source model that brings trouble like inserting image tag is only in teh paid version but that is possible to work around it with docxtemplater-image-module (see this discussion https://github.com/pwndoc/pwndoc/issues/13).

Hope it can give you some ideas / examples.

**Comment by noraj** [comment]
unstale
